bundles/org.eclipse.orion.client.ui/web/orion/widgets/projects/RunBar.js

```diff
@@ -216,13 +223,20 @@ define([
 					var dropdownMenuItemSpan = lib.$(".dropdownMenuItem", createNewItem); //$NON-NLS-0$
 					dropdownMenuItemSpan.classList.add("addNewMenuItem"); //$NON-NLS-0$
 					
-					var defaultDeployCommand = this._projectCommands.getDeployProjectCommands(this._commandRegistry)[0];
-					if (defaultDeployCommand) {
-						this._commandRegistry.registerCommandContribution(createNewItem.id, defaultDeployCommand.id, 1); //$NON-NLS-0$
+					var defaultDeployCommands = this._projectCommands.getDeployProjectCommands(this._commandRegistry);
+					if (defaultDeployCommands) {
+						for (var i = 0; i < defaultDeployCommands.length; i++) {
+							this._commandRegistry.registerCommandContribution(createNewItem.id, defaultDeployCommands[i].id, i + 1); //$NON-NLS-0$
+						}
 						domNodeWrapperList = [];
 						this._commandRegistry.renderCommands(createNewItem.id, dropdownMenuItemSpan, this._project, this, "button", null, domNodeWrapperList); //$NON-NLS-0$
-						domNodeWrapperList[0].domNode.textContent = "+"; //$NON-NLS-0$
-						this._setNodeTooltip(domNodeWrapperList[0].domNode, messages["createNewTooltip"]); //$NON-NLS-0$
+						for (var i = 0; i < defaultDeployCommands.length; i++) {
+							domNodeWrapperList[i].domNode.textContent = defaultDeployCommands[i].name;
+							this._setNodeTooltip(domNodeWrapperList[i].domNode, defaultDeployCommands[i].tooltip);
+						}
+						if (defaultDeployCommands.length === 1) {
+							domNodeWrapperList[0].domNode.textContent = "+";
```

Comment:
Here I hard coded that if there is only one deploy service (which is CF by default), the "new configuration" button will simply show "+" instead of its full name ("+CF" as I changed).
This keeps the consistency if debugger is disabled, but I think it's better to show "+CF" to tell the user that this is the configuration for a CF app.